cfme/tests/integration/test_cfme_auth.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from cfme.configure.access_control import Group, User
+from cfme import Credential, login
+from utils.conf import cfme_data
+
+
+def pytest_generate_tests(metafunc):
+    auth_modes = cfme_data['auth_test_data']
+    argvalues = [[authmode] for authmode in auth_modes]
+    idlist = [authmode for authmode in auth_modes]
+    if 'configure_auth' in metafunc.fixturenames:
+        metafunc.parametrize(['auth_mode'], argvalues, ids=idlist)
+
+
+@pytest.fixture()
+def data(request, auth_mode, add_group):
+    auth_data = cfme_data['auth_test_data'].get(auth_mode, {})
+    if add_group == 'evm_default_group':
+        auth_data['get_groups'] = False
+    return auth_data
+
+
+@pytest.fixture()
+def group(request, data, auth_mode, add_group):
+    if not data:
+        pytest.skip("No data spcified for user group")
+    credentials = Credential(
+        principal=data["username"],
+        secret=data["password"],
+    )
+    user_group = Group(description=data['group_name'], role="EvmRole-user",
+                       user_to_lookup=data["username"], ldap_credentials=credentials)
+    if add_group == 'retrieve_group':
```

Comment:
These strings like retrieve_group and create_group and such shall become constants to prevent typos.